### ISSUE TITLE
The code just stopped building, which is weird.

### ISSUE THREAD

**Issue by denisandroid** [zombodb/pgx#66]
```
cargo pgx package -d
```

```
installing extension
     Copying control file to `target/debug/x-pg12/usr/share/postgresql/extension/x.control`
     Copying shared library to `target/debug/x-pg12/usr/lib/postgresql/x.so`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected identifier")', /home/denis/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-pgx-0.1.5/src/commands/schema.rs:238:46
stack backtrace:
   0:     0x5586c6a63a30 - std::backtrace_rs::backtrace::libunwind::trace::he85dfb3ae4206056
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/../../backtrace/src/backtrace/libunwind.rs:96
   1:     0x5586c6a63a30 - std::backtrace_rs::backtrace::trace_unsynchronized::h1ad28094d7b00c21
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/../../backtrace/src/backtrace/mod.rs:66
   2:     0x5586c6a63a30 - std::sys_common::backtrace::_print_fmt::h901b54610713cd21
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/sys_common/backtrace.rs:79
   3:     0x5586c6a63a30 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hb0ad78ee1571f7e0
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/sys_common/backtrace.rs:58
   4:     0x5586c6a8d48c - core::fmt::write::h1857a60b204f1b6a
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/core/src/fmt/mod.rs:1080
   5:     0x5586c6a5cb72 - std::io::Write::write_fmt::hf7b7d7b243f84a36
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/io/mod.rs:1516
   6:     0x5586c6a67afd - std::sys_common::backtrace::_print::hd093978a5287b8ff
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/sys_common/backtrace.rs:61
   7:     0x5586c6a67afd - std::sys_common::backtrace::print::h20f46787581d56d7
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/sys_common/backtrace.rs:48
   8:     0x5586c6a67afd - std::panicking::default_hook::{{closure}}::h486cbb4b82ffc357
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/panicking.rs:208
   9:     0x5586c6a677a8 - std::panicking::default_hook::h4190c9e3edd4d591
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/panicking.rs:227
  10:     0x5586c6a68206 - std::panicking::rust_panic_with_hook::h72e78719cdda225c
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/panicking.rs:581
  11:     0x5586c6a67d89 - std::panicking::begin_panic_handler::{{closure}}::h8bd07dbd34150a96
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/panicking.rs:484
  12:     0x5586c6a63ebc - std::sys_common::backtrace::__rust_end_short_backtrace::hdb6b3066ad29028a
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/sys_common/backtrace.rs:153
  13:     0x5586c6a67d49 - rust_begin_unwind
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/panicking.rs:483
  14:     0x5586c6a8b0c1 - core::panicking::panic_fmt::hb15d6f55e8472f62
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/core/src/panicking.rs:85
  15:     0x5586c6a8aee3 - core::option::expect_none_failed::he492a18657d97593
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/core/src/option.rs:1234
  16:     0x5586c67220ea - cargo_pgx::commands::schema::generate_schema::hda22616a3eebdd0e
  17:     0x5586c6750dc9 - cargo_pgx::commands::install::install_extension::hc96cc1e8a41ad02f
  18:     0x5586c677a431 - cargo_pgx::commands::package::package_extension::he8eaf96779f7ce2f
  19:     0x5586c673a984 - cargo_pgx::do_it::h8dd20313c2db29d9
  20:     0x5586c673961d - cargo_pgx::main::h9d681428f61d126f
  21:     0x5586c6768d13 - std::sys_common::backtrace::__rust_begin_short_backtrace::hd8bb6822f116bd37
  22:     0x5586c6768f9c - std::rt::lang_start::{{closure}}::he5f098332b39b742
  23:     0x5586c6a68690 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h5e15f43ce28787f7
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/core/src/ops/function.rs:259
  24:     0x5586c6a68690 - std::panicking::try::do_call::h764f1891769113e9
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/panicking.rs:381
  25:     0x5586c6a68690 - std::panicking::try::h0f2fe4fdd5e0ad19
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/panicking.rs:345
  26:     0x5586c6a68690 - std::panic::catch_unwind::h1634e40dd3a19f80
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/panic.rs:382
  27:     0x5586c6a68690 - std::rt::lang_start_internal::h8d4bc7b854bbd5fe
                               at /rustc/7f7a1cbfd3b55daee191247770627afab09eece2/library/std/src/rt.rs:51
  28:     0x5586c673c712 - main
  29:     0x7f115c205152 - __libc_start_main
  30:     0x5586c67130ce - _start
  31:                0x0 - <unknown>
```

How to solve this? It seems that nothing has been updated, except for its own code.

**Comment by eeeebbbbrrrr** [comment]
what version of `cargo-pgx` do you have installed and what version of the various pgx dependencies are you using in your extension crate?

**Comment by denisandroid** [comment]
```
cargo pgx --version
cargo-pgx 0.1.5
```

```
[dependencies]
pgx = "0.1.5"
lazy_static = "1.4.0"
toml = "0.5.7"
lexical-core = "0.7.4"

[dev-dependencies]
pgx-tests = "0.1.5"

[dependencies.rand]
version = "0.7.3"
features = ["small_rng"]
```

**Comment by eeeebbbbrrrr** [comment]
Then we're going to need to see your code.  You've likely found a bug with pgx's schema generator

**Comment by denisandroid** [comment]
strace
```
Copying shared library to `target/debug/x-hidden12/usr/lib/postgresql/x.so`
) = 98
openat(AT_FDCWD, "hidden/hidden-hiddenxt/x/target/debug/libx.so", O_RDONLY|O_CLOEXEC) = 3
statx(3, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0755, stx_size=26679136, ...}) = 0
openat(AT_FDCWD, "hidden/hidden-hiddenxt/x/target/debug/x-hidden12/usr/lib/postgresql/x.so", O_WRONLY|O_CREAT|O_TRUNC|O_CLOEXEC, 0100755) = 4
statx(4, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0755, stx_size=0, ...}) = 0
fchmod(4, 0100755)                      = 0
copy_file_range(3, NULL, 4, NULL, 18446744073709551615, 0) = 26679136
copy_file_range(3, NULL, 4, NULL, 18446744073682872479, 0) = 0
close(4)                                = 0
close(3)                                = 0
openat(AT_FDCWD, "./src", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = 3
fstat(3, {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
getdents64(3, /* 11 entries */, 32768)  = 368
statx(3, "hidden_types", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFDIR|0755, stx_size=4096, ...}) = 0
openat(AT_FDCWD, "./src/hidden_types", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = 4
fstat(4, {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
getdents64(4, /* 4 entries */, 32768)   = 120
statx(4, "from_datum_num.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=3066, ...}) = 0
statx(4, "mod.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=316, ...}) = 0
getdents64(4, /* 0 entries */, 32768)   = 0
statx(3, "rand_gen_file_name.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=3880, ...}) = 0
statx(3, "lib.old.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=7844, ...}) = 0
statx(3, "hidden", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFDIR|0755, stx_size=4096, ...}) = 0
openat(AT_FDCWD, "./src/hidden", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = 5
fstat(5, {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
getdents64(5, /* 12 entries */, 32768)  = 416
statx(5, "hidden_from_datum.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=1346, ...}) = 0
statx(5, "hidden_alloc_ptr.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=1871, ...}) = 0
statx(5, "hidden_relation.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=2745, ...}) = 0
statx(5, "hidden_type.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=4047, ...}) = 0
statx(5, "hidden_write_datum.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=993, ...}) = 0
statx(5, "hidden_box.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=1830, ...}) = 0
statx(5, "hidden_array.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=2934, ...}) = 0
statx(5, "fn_trigger.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=8354, ...}) = 0
statx(5, "fn_callinfo.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=2121, ...}) = 0
statx(5, "hidden_ptr.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=2885, ...}) = 0
getdents64(5, /* 0 entries */, 32768)   = 0
statx(3, "hiddencom_types.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=0, ...}) = 0
statx(3, "lib.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=1510, ...}) = 0
statx(3, "config.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=4572, ...}) = 0
statx(3, "hiddenx_patch", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFDIR|0755, stx_size=4096, ...}) = 0
openat(AT_FDCWD, "./src/hiddenx_patch", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = 6
fstat(6, {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
getdents64(6, /* 3 entries */, 32768)   = 88
statx(6, "macros_extern.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=1872, ...}) = 0
getdents64(6, /* 0 entries */, 32768)   = 0
statx(3, "hidden_mod.rs", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=6489, ...}) = 0
getdents64(3, /* 0 entries */, 32768)   = 0
openat(AT_FDCWD, ".", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = 7
fstat(7, {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
brk(0x556f7df09000)                     = 0x556f7df09000
getdents64(7, /* 10 entries */, 32768)  = 288
brk(0x556f7df01000)                     = 0x556f7df01000
close(7)                                = 0
openat(AT_FDCWD, "x.control", O_RDONLY|O_CLOEXEC) = 7
read(7, "comment = 'x:  Created by p"..., 8192) = 137
read(7, "", 8192)                       = 0
close(7)                                = 0
openat(AT_FDCWD, "./sql", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = 7
fstat(7, {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
getdents64(7, /* 3 entries */, 32768)   = 88
statx(7, "load-order.txt", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=21, ...}) = 0
getdents64(7, /* 0 entries */, 32768)   = 0
close(7)                                = 0
openat(AT_FDCWD, "./src/config.rs", O_RDONLY|O_CLOEXEC) = 7
statx(7, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=4572, ...}) = 0
read(7, "use std::{collections::HashMap, "..., 4573) = 4572
read(7, "", 1)                          = 0
close(7)                                = 0
futex(0x7fbc4290b080, FUTEX_WAKE_PRIVATE, 2147483647) = 0
brk(0x556f7df22000)                     = 0x556f7df22000
brk(0x556f7df43000)                     = 0x556f7df43000
brk(0x556f7df64000)                     = 0x556f7df64000
brk(0x556f7df85000)                     = 0x556f7df85000
brk(0x556f7dfa8000)                     = 0x556f7dfa8000
brk(0x556f7dfa6000)                     = 0x556f7dfa6000
brk(0x556f7dfa2000)                     = 0x556f7dfa2000
brk(0x556f7df9f000)                     = 0x556f7df9f000
statx(AT_FDCWD, "./sql/config.generated.sql", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7ffc466fe1b0) = -1 ENOENT (Нет такого файла или каталога)
openat(AT_FDCWD, "./src/hidden/fn_callinfo.rs", O_RDONLY|O_CLOEXEC) = 7
statx(7, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=2121, ...}) = 0
read(7, "\nuse crate::hidden_types::BigI"..., 2122) = 2121
read(7, "", 1)                          = 0
close(7)                                = 0
statx(AT_FDCWD, "./sql/hidden_fn_callinfo.generated.sql", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7ffc466fe1b0) = -1 ENOENT (Нет такого файла или каталога)
openat(AT_FDCWD, "./src/hidden/fn_trigger.rs", O_RDONLY|O_CLOEXEC) = 7
statx(7, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=8354, ...}) = 0
read(7, "\nuse hiddenx::AnyElement;\nuse hiddenx::P"..., 8355) = 8354
read(7, "", 1)                          = 0
close(7)                                = 0
brk(0x556f7dfc0000)                     = 0x556f7dfc0000
brk(0x556f7dfe1000)                     = 0x556f7dfe1000
statx(AT_FDCWD, "./sql/hidden_fn_trigger.generated.sql", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7ffc466fe1b0) = -1 ENOENT (Нет такого файла или каталога)
openat(AT_FDCWD, "./src/hidden_types/from_datum_num.rs", O_RDONLY|O_CLOEXEC) = 7
statx(7, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=3066, ...}) = 0
read(7, "\nuse crate::hidden::hidden_write_d"..., 3067) = 3066
read(7, "", 1)                          = 0
close(7)                                = 0
statx(AT_FDCWD, "./sql/hidden_types_from_datum_num.generated.sql", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7ffc466fe1b0) = -1 ENOENT (Нет такого файла или каталога)
openat(AT_FDCWD, "./src/lib.old.rs", O_RDONLY|O_CLOEXEC) = 7
statx(7, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=7844, ...}) = 0
read(7, "#[macro_use] extern crate hiddenxr_1"..., 7845) = 7844
read(7, "", 1)                          = 0
close(7)                                = 0
statx(AT_FDCWD, "./sql/lib.old.generated.sql", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7ffc466fe1b0) = -1 ENOENT (Нет такого файла или каталога)
openat(AT_FDCWD, "./src/lib.rs", O_RDONLY|O_CLOEXEC) = 7
statx(7, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=1510, ...}) = 0
read(7, "\nhidden_module_magic!();\n \nuse hiddenx::"..., 1511) = 1510
read(7, "", 1)                          = 0
close(7)                                = 0
statx(AT_FDCWD, "./sql/lib.generated.sql", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7ffc466fe1b0) = -1 ENOENT (Нет такого файла или каталога)
openat(AT_FDCWD, "./src/hiddenx_patch/macros_extern.rs", O_RDONLY|O_CLOEXEC) = 7
statx(7, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=1872, ...}) = 0
read(7, "\n/*\n\n/*\n\t\320\227\320\260\321\207\320\265\320\274 \320\264\320\260\320\275\320\275\321\213\320\271"..., 1873) = 1872
read(7, "", 1)                          = 0
close(7)                                = 0
statx(AT_FDCWD, "./sql/hiddenx_patch_macros_extern.generated.sql", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7ffc466fe1b0) = -1 ENOENT (Нет такого файла или каталога)
openat(AT_FDCWD, "./src/hidden_types/mod.rs", O_RDONLY|O_CLOEXEC) = 7
statx(7, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=316, ...}) = 0
read(7, "\nuse crate::hidden::hidden_type::P"..., 317) = 316
read(7, "", 1)                          = 0
close(7)                                = 0
statx(AT_FDCWD, "./sql/hidden_types_mod.generated.sql", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7ffc466fe1b0) = -1 ENOENT (Нет такого файла или каталога)
openat(AT_FDCWD, "./src/hidden/hidden_alloc_ptr.rs", O_RDONLY|O_CLOEXEC) = 7
statx(7, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=1871, ...}) = 0
read(7, "\n\nuse crate::hidden::hidden_ptr::C"..., 1872) = 1871
read(7, "", 1)                          = 0
close(7)                                = 0
statx(AT_FDCWD, "./sql/hidden_hidden_alloc_ptr.generated.sql", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7ffc466fe1b0) = -1 ENOENT (Нет такого файла или каталога)
openat(AT_FDCWD, "./src/hidden/hidden_array.rs", O_RDONLY|O_CLOEXEC) = 7
statx(7, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=2934, ...}) = 0
read(7, "\nuse crate::hidden::hidden_ptr::CP"..., 2935) = 2934
read(7, "", 1)                          = 0
close(7)                                = 0
statx(AT_FDCWD, "./sql/hidden_hidden_array.generated.sql", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7ffc466fe1b0) = -1 ENOENT (Нет такого файла или каталога)
openat(AT_FDCWD, "./src/hidden/hidden_box.rs", O_RDONLY|O_CLOEXEC) = 7
statx(7, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=1830, ...}) = 0
read(7, "\nuse crate::hidden::hidden_ptr::CP"..., 1831) = 1830
read(7, "", 1)                          = 0
close(7)                                = 0
statx(AT_FDCWD, "./sql/hidden_hidden_box.generated.sql", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7ffc466fe1b0) = -1 ENOENT (Нет такого файла или каталога)
openat(AT_FDCWD, "./src/hidden/hidden_from_datum.rs", O_RDONLY|O_CLOEXEC) = 7
statx(7, "", AT_STATX_SYNC_AS_STAT|AT_EMPTY_PATH, STATX_ALL, {stx_mask=STATX_ALL|0x1000, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=1346, ...}) = 0
read(7, "\nuse crate::hidden::hidden_type::P"..., 1347) = 1346
read(7, "", 1)                          = 0
close(7)                                = 0
write(2, "thread '", 8thread ')                 = 8
write(2, "main", 4main)                     = 4
write(2, "' panicked at '", 15' panicked at ')         = 15
write(2, "called `Result::unwrap()` on an "..., 73called `Result::unwrap()` on an `Err` value: Error("expected identifier")) = 73
write(2, "', ", 3', )                      = 3
write(2, "/home/denis/.cargo/registry/src/"..., 98/home/denis/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-hiddenx-0.1.5/src/commands/schema.rs) = 98
write(2, ":", 1:)                        = 1
write(2, "238", 3238)                      = 3
write(2, ":", 1:)                        = 1
write(2, "46", 246)                       = 2
write(2, "\n", 1
)                       = 1
write(2, "note: run with `RUST_BACKTRACE=1"..., 78note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
) = 78
close(6)                                = 0
close(4)                                = 0
close(5)                                = 0
close(3)                                = 0
sigaltstack({ss_sp=NULL, ss_flags=SS_DISABLE, ss_size=8192}, NULL) = 0
munmap(0x7fbc42cc3000, 12288)           = 0
exit_group(101)                
```

Something very strange

**Comment by eeeebbbbrrrr** [comment]
that's not helpful here at all.  lets see your `.rs` files...

**Comment by denisandroid** [comment]
This is as weird as **possible**.

# 1 code
```rust
pub trait PgSType {}

impl<T> PgSType for T {}

pub trait FromDatum: where Self: PgSType {
	
}
```

cargo check
```
Ok.
```
cargo pgx -d

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected identifier")', /home/denis/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-pgx-0.1.5/src/commands/schema.rs:238:46

```


# 2 code

```rust
pub trait PgSType {}

impl<T> PgSType for T {}

pub trait FromDatum where Self: PgSType {
	
}
```

cargo check
```
Ok.
```

cargo pgx -d
```
Ok.
```

**Comment by eeeebbbbrrrr** [comment]
Dunno how to answer this.  The error from schema.rs:238 is coming from the `syn` package which parses code.

Perhaps what happened is they released a breaking change that they tagged as semver-compatible and your system picked that up.  I've seen that before with syn, quote, and bingen.

I'll have to look into it when I have some time.

**Comment by eeeebbbbrrrr** [comment]
should be fixed in 0.1.13.

Hopefully problems like this will no longer occur as I've added Cargo.lock to the project.
